src/framework/Fn.ts

```
@@ -22,125 +42,145 @@ type MethodType<T, K extends keyof T> = T[K] extends (...args: any[]) => any
-   * @example trackFn.wasCalledTimes(3);
+   * Replaces the tracked function with a custom implementation
+   * @param {Function} fn - The custom function to be used
+   * @returns {TrackFn} The updated tracked function with the custom implementation
+   * 
+   * @example
+   * const add = (a, b) => a + b
+   * const trackFn = Fn(add)
+   * trackFn.use((a, b) => a * b)
    */
-  wasCalledTimes(n: number): boolean;
+  use(fn: Function): TrackFn
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)